app/controllers/api/v1/habits_controller.rb

```diff
@@ -12,15 +12,33 @@ def index
   end
 
   def create
-    habit = Habit.new(habit_params)
-    if habit.save
-      flash[:notice] = "Habit Successfully Created"
-      flash[:color]= "valid"
+    @goal = Goal.where(title: habit_params[:schedules_attributes]['0'][:goal]).first
+    @habit = Habit.new(
+      title: habit_params[:title],
+      description: habit_params[:description], goal: @goal
+    )
+    if @habit.save
+      @schedules = []
+      habit_params[:schedules_attributes].each do |key, value|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.